### ISSUE TITLE
Cheese Ore very difficult to find vs. OreGen parameters ?

### ISSUE THREAD

**Issue by fira** [micdoodle8/Galacticraft#3184]
Minecraft version: 1.11.2
Galacticraft version: 4.0.1.128 with inital worldgen in 4.0.1.120
Forge version: 13.20.1.2588
Type: SMP

Galacticraft Add-Ons: Planets, ExtraPlanets added after later on
Other Mods: RFTools+Dimensions, COFH Thermal Expansion/Foundation/Dynamics, AE2, other stuff

We're still discovering GC and recently started making AstroMiners and attempted to get some extra Cheese Ore to build more. It kind of hit me how scarce it seems to be. Maybe it's just bad luck, maybe i just don't understand oregen, maybe not. Could it be possible to check ?

After checking in Wiki and in worldgen parameters, compare the following for example for around 10K Moon Stone blocs mined:
 * Copper ore: 26 clumps per chunk between y=0 and y=60, 60 blocs found and many more unmined all around
 * Tin ore: 23 clumps per chunk between y=0 and y=60, 39 blocs found and many more unmined all around
 * Cheese ore: 12 clumps per chunk between y=0 and y=128, 2 blocs found

A friend tried to excavate a whole chunk (~20K blocs from Y=80 to Y=0) and only found two aswell.
Is it really meant to be that difficult to find, or could that be an issue ? Sorry if it's not one.

Thanks !

**Comment by radfast** [comment]
Hi fira, thank you for the issue report.  I'll check the worldgen, maybe make some test worlds where it logs the position of each cheese ore which is supposed to be there so I can check.

It's good to hear from a player group who are playing through Galacticraft's progression from beginning to end.  If you have built AstroMiners you have done well.  Please do give any more comments you have about gameplay, balance, fairness etc, I appreciate these.

Known issues:
- At the start of the game the player is limited by the amount of mining.  A lot of metals, redstone, etc are needed to get started, to make a full set of machines, and to build your first Tier 1 Rocket.  This difficulty is **intended**.  You need to be quite advanced in the game to build a rocket, it's not something for the first 15 minutes.  There is emotional payback this way - reaching the Moon feels like a big achievement.
- If you are playing with TE there is plenty of copper and tin, and your iron and aluminium production is increased by TE's machines.  So starting out should be OK.
- You can become stranded on the Moon or have a hard time there if you don't take everything necessary with you.  Again **intended**.  The Moon is a dangerous place!  More accurately it's a person's lack of experience of it and preparation which are dangerous ...
- Meteoric Iron is slow to obtain, because you have to find meteors.
- There is an enforced linear progression:  Tier 1 Rocket -> Moon Dungeon -> Tier 2 Rocket -> Mars Dungeon -> Tier 3 Rocket -> Astrominer.   I'm not so happy with this - I prefer the game to be non-linear.   Perhaps one idea can be an alternate recipe for Orion Drive using a Venus ore?

**Comment by fira** [comment]
We're about a dozen friends playing Minecraft occasionnaly. Overall I think everyone here's in agreement to say GC is very polished. We had a few gamebreaking bugs or issues, but hopefully nothing we couldn't fix through adminning or just determination. 

Since you seem eager to get some feedback, let me give some :D 

 * For context, we used RFtool TPs only as a way to get back in emergency cases, but decided to keep using Galacticraft rockets for any kind of space travel. We've yet to use Automated Cargo Rockets. Also, we eventually used IC2 Jetpacks which somewhat trivialised part of the content (esp. T2 Boss and Asteroids).
 * Getting stranded is fine I think, in fact that's how I got into GC. One of my friends kept going to next tier without proper preparation and got stranded on every single planet, forcing me to come rescue him every time by following him closely in the tech tree. Good times.
 * However, being stranded after dying was very, very frustrating for some players for several reasons. They'd get stuck in dying loops due to respawning on the planet without survival gear. Oddly enough, we also encountered something strange which is that after dying a few time and respawning at Landing position, people would eventually spawn at x=0,y=0 and just die there in a loop. 
 * /dimensiontp is great, but it opens the possibility for people to just create Space Stations or go anywhere. It'd be great to have a different command to send a player directly to a specified dimension. I ended up having to modify player data files a few times to change their dimensions.
 * For the above two reasons, I ended up disabling the respawn-in-dimension in Galacticraft config, which is on by default. Ideally, I think there should be a way to give players the choice, if at all possible.
 * Even then, some of the least resilient and unprepared players eventually gave up due to having to recraft Rockets every time they'd die. I'm not sure how this could be implemented in gameplay, but it could help team expeditions to have a way to recraft them faster.
 * Out of several dozen launchs, we've had three times an odd issue with players reporting that they wouldn't get their Launch Pad back in the lander, and got stranded for that reason. Not sure if it's an actual bug or just mistakes...
 * No big issue, but it was rather confusing at first that Schematics are actually bound to player and not workbench. This forced us to farm dungeons on some planets as we were a few to want to get to next tier. Would be great to be a togglable (unless i missed it?)
 * I've yet to get to the Asteroids Dimension myself, but players that got there so far were really impressed by the amount of work put into it, with different kinds of asteroids, derelicts space stations, and such.
 * Hooraaaay for space stations ! I was personally building a Sky Base on the Overworld before finding out we could make Space Stations. As you can imagine, that changed my plans heavily :P. Takes a while getting used to Zero-G (and learning not to carry a jetpack or you just get shot off into space), but it's very enjoyable. I still want to see the station rotation thrusters in action...
 * I assume this is a technical limitation, but it is annoying that Sealers can only seal a very small space. This prevents big rooms with machinery, so much in fact that i ended up deciding to build an unpressurized part of the station for that. Another player tried to seal a village hut and had to partially fill it as it was too big.
 * My 2cents on third party expansions: ExtraPlanets kinda sucks. I mean, well, that's harsh. It doesn't "suck". It's well thought and has lots of extra content. But when you're used to the polishing quality of Galacticraft and suddenly end up in ExtraPlanets territory, suddenly there's a lot more grind and extra mechanics (such as pressure and radiations) that are unpractical or counterintuitive (vs GC vanilla), and didn't make it very fun. Then again, the target was set pretty high by GC Vanilla :P 
 * Because of the amount of ressources needed in ExtraPlanets, i started to automate part of the GC vanilla crafting with AE2. This turned out to be a major annoyance with the Compressor: AE2 simply doesn't seem to handle machines with freeform patterns properly. It'd start the crafting properly, but sometimes add ressources during the process as they become available, and make unbalanced stacks in the Compressed plates recipes for example.. Also, parallel crafting with it doesn't work at all : AE2 sees free space in the Compressor and throws more stuff in it. This ends up with situations where the Compressor is filled to the brim with random stuff and you have to unclog it yourself and relaunch crafts.

We probably made a lot of mistakes, but had a lot of fun so far with GC. Thanks a lot : )

**Comment by radfast** [comment]
thank you this is very valuable

I'll respond to one group of issues immediately, and the rest in stages

> * However, being stranded after dying was very, very frustrating for some players for several reasons. They'd get stuck in dying loops due to respawning on the planet without survival gear. Oddly enough, we also encountered something strange which is that after dying a few time and respawning at Landing position, people would eventually spawn at x=0,y=0 and just die there in a loop.
* /dimensiontp is great, but it opens the possibility for people to just create Space Stations or go anywhere. It'd be great to have a different command to send a player directly to a specified dimension. I ended up having to modify player data files a few times to change their dimensions.
* For the above two reasons, I ended up disabling the respawn-in-dimension in Galacticraft config, which is on by default. Ideally, I think there should be a way to give players the choice, if at all possible.
* Even then, some of the least resilient and unprepared players eventually gave up due to having to recraft Rockets every time they'd die. I'm not sure how this could be implemented in gameplay, but it could help team expeditions to have a way to recraft them faster.

1. The server admin should be able to use `/dimensiontp` to move one specific player, using `/dimensiontp <name>`.  Command can be run from server console or in game.  Sure, the player gets taken to the Celestial Selection screen not directly to the planet, but they should then click Overworld to return home - they'll spawn high, presumably without a parachute, and then either splash down in water or die on the Overworld.

2. I can see one improvement to make - the teleported player should automatically be given a parachute, because we don't want the poor guy dying yet again!

3. Additionally, we could easily add a new command where the player can decide to give up on space and take themselves back to the Overworld, losing all items.  Command needs a name ... I'm thinking `/houstonwehaveaproblem`, and you'd have to type it twice to confirm.  I'm not sure how the type of player who gets themselves into this situation - i.e. someone who does not read the guidance on the Wiki etc - is going to **know** about that command.  I guess the game could detect a looped dying pattern and prompt for that command in chat.  (Don't want to automate it totally, because the looped dying could be for another reason, e.g. a player attack on a person's Moon home, on a griefing/raiding server).

4. Big servers with lots of Sponge plugins manage these types of issues using Essentials (or **Forge Essentials** is an option if you don't want Sponge).  Every player can use `/spawn`, `/home` and `/back` commands, at least one of which will probably take you back to the Overworld without any item loss.

5. Any more help you can give me on the situation causing the respawn pattern switching to x=0,z=0?  That sounds like a bug.  How many times roughly to die before that happened?  It will maybe be relevant to (6).

6. A block I plan to add next is a **Space Emergency Kit**.  This would be a cabinet positioned by your respawn location.  Inside is an Oxygen Mast, Gear, Light Oxygen tank.  Maybe also thermal padding, 4 glowstone torches and a tin of food?  And a single click on it will equip all the oxygen items.  Normally the player should have to craft the contained items and then craft the emergency kit himself, but again we could add a server command to allow the admin to auto-spawn one of these.

7. I don't follow what you are saying about rocket loss for players who die.  The rocket will still be on the Moon (or other dimension), in the lander where the player landed.  It's not lost, it's just inaccessible to that player.  But another player could go and find it and bring it to him - or he could "borrow" a rocket from his friend, go and retrieve it, and come home with 2 rockets.  (Not sure I would lend my rocket to that guy but still...)

8.  For a player who is only "slightly" dying - e.g. well equipped generally but dies to mobs while mining on the Moon or something like that - then the emergency pack at (6) will help not get into a dying loop. 

9.  If the player crash lands on the Moon then the rocket item will drop, and it may be quite hard to find in all the debris at the crash landing site.  Like any item on a Minecraft server it will despawn after 5 minutes but there's still a chance to go and get it - or a smart player would go offline, wait until his friend is playing, then go back to that spot together with a friend.

10. If the player is flying to the Moon with no oxygen equipment, crash landing, and generally doing some really dumb stuff then I think it's OK for that player to lose his rocket.  But I agree with your general points - I would like to give more possibilities for help for the players who are **trying** but dying due to bad luck.  The Alien Villagers trading sapphires for Oxygen Masks etc was _supposed_ to help with this type of thing, but actually it doesn't do anything at all for a player who has no Frequency Module, or a player who is dying before he can mine any sapphires.

TL;DR:  I'll add a command so players can respawn themselves on the Overworld without admin assistance.  And a craftable **Space Emergency Kit**.

**Comment by radfast** [comment]
> it was rather confusing at first that Schematics are actually bound to player and not workbench. This forced us to farm dungeons on some planets as we were a few to want to get to next tier. Would be great to be a togglable (unless i missed it?)

There's no toggle. This is micdoodle8's design, it's part of the space race concept: each player individually is advancing through the tiers (though teamwork can obviously help with that).  It also prevents loss of valuable progress if a NASA workbench is destroyed - even if the workbench dropped schematic items when destroyed, those could be lost to lava, they could fall out of the world on Asteroids, or they could simply despawn after 5 minutes etc.  I'm not sure we would change this.  Moon dungeons are built to be farmable, the dungeon boss should reset after around 10-15 minutes.

> I've yet to get to the Asteroids Dimension myself, but players that got there so far were really impressed by the amount of work put into it, with different kinds of asteroids, derelicts space stations, and such.

Thank you :)   https://wiki.micdoodle8.com/wiki/Abandoned_Base was 1 month's work.  The hangar design is by the awesome @SpacedViking.  The room design is by me.

> it is annoying that Sealers can only seal a very small space. This prevents big rooms with machinery, so much in fact that i ended up deciding to build an unpressurized part of the station for that. Another player tried to seal a village hut and had to partially fill it as it was too big.

https://wiki.micdoodle8.com/wiki/How_To#How_to_fill_a_room_with_Breathable_Air

Use 2 sealers ... or 20 ... to seal larger volumes.  They work in concert with each other (in parallel if you're an electronics guy).  There's no limit, other than the oxygen and energy needed to keep them working.  I've personally tested 80 sealers in a 100,000 block volume with no issues.  Much more than that could cause a momentary server lag, if the space switches between sealed and unsealed.


Some other things:

https://wiki.micdoodle8.com/wiki/How_To#How_to_move_on_Space_Stations

Compressor automation with AE2 - this is possible, see https://wiki.micdoodle8.com/wiki/Electric_Compressor#Automation end notes for AE2

**Comment by TheQuadShot** [comment]
I like the Idea of a "rescue/emergency" chest/block. I do this on my own though as first step in my build process anywhere I land. I setup a bubble gen near the (within 3 blocks) chest and power it with solar or battery and lever to activate bubble gen so there is a small "safe haven" to get re-equipped in and scatter them around my landing zone/space station (space station is always a very large torus with domes/platforms in the center and/or on the ordinals of the torus (https://imgur.com/ajMs0nq) and it is always pressurised, eventually.) 

Is it possible to have the schematics shared to each member of the Space Race Team?

Also as a small "nit" the quote by Jack Swigert was "Uh, Houston, we've had a problem", "Houston we have a problem" was Tom Hanks....

@fira  When building your space station use the Grapple!

**Comment by radfast** [comment]
>  Is it possible to have the schematics shared to each member of the Space Race Team?

Nice solution, we can do that.  That shouldn't disturb the game balance too much, it eliminates the grind, and it provides a really worthwhile bonus for joining a Space Race Team.

> Also as a small "nit" the quote by Jack Swigert was "Uh, Houston, we've had a problem", "Houston we have a problem" was Tom Hanks....

lol I know this, I <3 Apollo Space Program history.  But the popular meme these days is the Tom Hanks version.  And the real version is too long to type...  We can put info about the real version on our Wiki page for this command.

**Comment by TheQuadShot** [comment]
@radfast  Nice! I also was a very big NASA nerd and when I was a much younger man I had to get NASA Certified to build circuits/components that were bound for space. So I was able to goto several Space Science symposiums and got to meet lots of people from that era of our space program. (cut to scene of engineers around a table to "solve" Apollo 13's problem) ;-)

**Comment by radfast** [comment]
ok I think you win, that's real engineering!

**Comment by TheQuadShot** [comment]
lol I was/am the guy that wants to know the "How" of things. I would of liked to meet some of those great astronauts and cosmonauts. I did get to meet some russian scientists and professors and even traded postage stamps with some of them for a few years.

**Comment by radfast** [comment]
Space Emergency Kit item is now in the game.  Using `/gckit <name>` an admin can give one to a player.

Here's the recipe.  The items the player receives (when he uses the kit) exactly match the recipe.  So if there's an accidental use, the player can re-craft the kit from the items.  The oxygen items will auto-equip to the correct slots when the kit is used.
![ekit](https://user-images.githubusercontent.com/6842258/36645525-1daed660-1a62-11e8-9c0f-b08981f1f217.png)

I'm now looking at a dispenser box to hold **one** of these.  Or, players anticipating lots of emergencies could fill a regular Minecraft Dispenser with lots of these ...

**Comment by TheQuadShot** [comment]
I like it! Can the food item be any of the GC food items?

**Comment by radfast** [comment]
I had to choose one specific food can for the recipe, so here it's apple.  **Maybe I should make it potato in homage to [The Martian](https://en.wikipedia.org/wiki/The_Martian_(Weir_novel))**?

Likewise, the parachute must be red and the potions must be the specific kinds.

An add-on could add additional Emergency Kits with different recipes and contents but similar functionality overall.

**Comment by EzerArch** [comment]
> Additionally, we could easily add a new command where the player can decide to give up on space and take themselves back to the Overworld, losing all items. (...)

In vanilla, what happens if I die in The Nether or The End? I respawn next to my bed in the Overworld. Then I use/craft a spare set of armor, weapons and food and go to rescue my drops and continue my task.

Galacticraft should follow the same logic: if you, for example, crash and die with the Lander during the moon descent stage, you'll respawn on the Overworld and start over your mission that you failed. It may seem penalizing but it's no different from what happens in vanilla -- and BTW it's better than being stuck in the dying loops in space.

Additionally, the Cryogenic Chamber can be used to move your respawn point from the Overworld to a safer place (assuming the player will build a base and place the CryoChamber in there).

**Comment by radfast** [comment]
Not sure about that, @EzerArch 
- If you get force-moved back to your Overworld bed, you will lose access to your lander with rocket, launch pads and fuel ... seems too harsh.
- Players can easily die before they built a bed / Cryogenic Chamber on the Moon, including from mob attacks if they land on the Moon in the Lunar night.
- So long as you die close to your lander (where you respawn), a good player has enough time to gather up items and survive.
- Even if you can't find all the items immediately (for example Oxygen Gear is missing) you can get most of them, move them to a better spot close to your respawn point, then get those ones back quickly next time it loops, giving you enough time to go search for the final missing item.  You have around 1 minute.
- The **problem** situation is when the player dies far from his lander - then there is no time to run back to the death location to gather the items.  So players who do not have redundant backup resources should **not** move too far from the lander.  I find this realistic :)
- This** is where the **Space Emergency Kit** becomes useful.  The planned Space Emergency Box should be placed down close to your lander, so you have easy access to it when you respawn.
Basically**** it must be possible to survive it, unless items despawned or fell into lava (but there is no lava on the Moon).

**Comment by radfast** [comment]
Next build there will be more Cheese Ore, approximately twice as much and more chance for the clumps to be 2 or 3 ore blocks.  Similarly, the clumps of other valuable ores on Moon, Mars, Venus etc will now be more consistent sizes - there was a bug where the clumps sometimes ended up smaller than intended, so that 1 or 2 block clumps were quite common even for the metal ores.

**Comment by radfast** [comment]
Now added a `/gchouston` command.

If the player is dying a lot, the in-game chat sequence is like this:

>    Uh, Houston, we've had a problem....  (Type /gchouston to abandon mission and return home.)

Then if the command is used:

>    This is Mission Control, say again please?  (To abandon mission, type /gchouston again to confirm.)

>    OK, stand by, <name>, teleporting to Overworld home position.

**Comment by radfast** [comment]
The server admin can also use the command without repetition, by typing `/gchouston \<name\>`

**Comment by radfast** [comment]
I have added the suggestion that unlocking a Schematic (in a NASA workbench) unlocks the same schematic for all members of the same Space Race team.

I think that the gameplay improvements suggested in this thread are now all implemented, so I will close.

Thank you for all contributions, this will make a real difference to the enjoyment of the game for players in future.
